### PR TITLE
FIX Register component now handles validation errors, and resets errors between methods

### LINKED ISSUE #73: Base registration component handler doesn't care about non-200 results
If my MFA component's backend RegisterHandlerInterface implementation throws an exception during the `register()` method, the result is a 400 response, but the frontend components don't check for it and continue on to generating backup codes. This leaves the user registered with only back up codes. It should return an error on the frontend instead and allow the process to continue.

## Acceptance criteria

* Throwing an `AuthenticationFailedException` (doesn't exist in master at time of writing) from a `RegisterHandler::register()` method will be caught by the MFA frontend app
* As above, will not proceed to generation of backup codes
* As above, will pass the exception's message down to React registration components as a prop (e.g. `{ hasError: true, errorMessage: $ex->getMessage() }` (pseudo code - example only)

### LINKED ISSUE #80: Error state passed to login components is not unique
Example:

* Register TOTP as your method
* Login with it and enter an incorrect code
* Choose More options
* Select backup codes

Observe error state on the backup codes component. This should be reset when moving out of the login component.

### PR DESCRIPTION
**Pull request by robbieaverill**
Fixes #73 and fixes #80

### CODE REVIEW

**Comment by robbieaverill** [comment]
Sorry about the noise - eslint was moaning about the order I had methods in so I had to move them around a bit

**Comment by robbieaverill** [comment]
I guess I should write tests for this. 20% of patch coverage is a poor effort.

File: client/src/components/Register.js [hunk 1]
```diff
@@ -134,27 +136,67 @@ class Register extends Component {
...
-      .then(() => {
-        // If there's a backup method that's not registered then we initialise that
-        if (
-          this.shouldSetupBackupMethod()
-          && selectedMethod.urlSegment !== backupMethod.urlSegment
-        ) {
+      .then(response => {
+        // Failure states are captured here
+
+        // Expected structure: ['Registration failed', 'Because you misspelled your password.']
```
Comment (ScopeyNZ) [inline client/src/components/Register.js]: Maybe we can make the server side have a more declarative structure? I'm not sure I like "assume the second item in the array is the thing we want".

**Comment by robbieaverill** [comment]
I agree those suggestions sound good, but could be handled separately

File: client/src/components/Register.js (hunk 1 above)
Comment (robbieaverill) [reply client/src/components/Register.js]: Yeah, the other places where `errors` is returned from the backend is pushing single string messages into its array. This particular case is returning two strings in the array, so it's different to the usual response format. I'll change it to match the rest.

Comment (robbieaverill) [reply client/src/components/Register.js]: I've made it just return the error message from the authenticator, not the "Registration failure" message as well

**Pull request merged**